### ISSUE TITLE
tools/install-redhat.sh should be tools/install-rhel.sh

### ISSUE THREAD

**Issue by nkadel** [PowerShell/PowerShell#20752]
### Summary of the new feature / enhancement

"redhat" is the name of a very obsolete set of Red Hat published operating systems. The newer ones are called "rhel", and tools/install-redhat.sh should be renamed to reflect that.

It is also hard-coded to RHEL 7, which will be obsolute in June 2024. Ideally, it should install a .repo speciric to the "releasever", and publish the .repo file to /etc/yum.repos.d/packages-microsoft-com-prod.repo file rather than the confusingly short "prod.repo".

**Comment by rhubarb-geek-nz** [comment]
Rather than installing the repo file directly can you not install https://packages.microsoft.com/config/rhel/9.0/packages-microsoft-prod.rpm directly, where `rhel` and `9.0` have come from `/etc/os-release` ?

Yum can install directly from a URL so there is no need for curl

It has much boiler plate code looking for AIX, Windows, SunOS which is all redundant

It only needs to look for a file marked /etc/os-release and look for ( for example )

```
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="9.3"
```

And confirm that either ID or ID_LIKE contain 'rhel', 'centos' or 'fedora' to determine if an RPM based system.  

It does not even have to do that, all it has to confirm is that an RPM for the repo exists at the right URL at https://packages.microsoft.com/config/$ID/$VERSION_ID/packages-microsoft-prod.rpm.

**Comment by nkadel** [comment]
Let's fix the simple problems first, a confusing and inaccurate name for the installpsh-redhat.sh script.

Interpreting /etc/os-release correctly is a support nightmare. And for the example you point out, naming the .repo file "mariner-microsoft.repo" when the actual repo is "mariner-official-microsoft" is another level of inconsistent foolish that will one day require cleanup to address correctly. That kind of thing is why I suggested only one or two changes, and no re-arrangement of published files.

The problem with renaming .repo files is that if you run the install script later, and publish a different file with the same repo name, you break yum for everyone by having duplicate named repos. Hilarity and chaos ensue and people yelling at you because you broke their deployment tools. Renaming .repo files should be done only very, very carefully, just as interpreting /etc/os-release entries is a complete crap shoot.

**Comment by nkadel** [comment]
Yum can install RPMs directly from a URL. But installing RPM's requires knowledge of the URL for the valid release RPM, which is extra knowledge of the upstream vendor to maintain a consistent release URL. It's an extra level of trust, and has *zero* to do with the much smaller and safer feature request I've made.

**Comment by rhubarb-geek-nz** [comment]
> The problem with renaming .repo files is that if you run the install script later, and publish a different file with the same repo name, you break yum for everyone by having duplicate named repos. 

Agreed, these config files should never be added to a system with arbitrary scripts, they should be installed using an rpm which takes ownership of the file and is responsible for updating it and even changing the name if necessary.

For instance the script does not check if there is already and existing repo file for that repository or if the official microsoft rpm has already been installed.

You might have been better with the script doing

```
yum install https://github.com/PowerShell/PowerShell/releases/download/v7.4.0/powershell-7.4.0-1.rh.x86_64.rpm
```
and it will sort out the dependencies.

**Comment by rhubarb-geek-nz** [comment]
> Let's fix the simple problems first, a confusing and inaccurate name for the installpsh-redhat.sh script.

```
$ rpm -qilp packages-microsoft-prod.rpm
Name        : packages-microsoft-prod
Version     : 1.0
Release     : 1
Architecture: noarch
Install Date: (not installed)
Group       : Applications/Internet
Size        : 206
License     : GPLv2
Signature   : RSA/SHA256, Tue May  3 03:55:59 2022, Key ID eb3e94adbe1229cf
Source RPM  : packages-microsoft-prod-1.0-1.src.rpm
Build Date  : Tue May  3 03:55:50 2022
Build Host  : sark-apt-wus
URL         : http://redhat.com
Summary     : Configure clients for the Microsoft prod repo
Description :
Configure clients for the Microsoft prod repo
/etc/yum.repos.d/microsoft-prod.repo
```

So the name should be `/etc/yum.repos.d/microsoft-prod.repo` to be consistent, and if it is already there then don't overwrite it.

**Comment by nkadel** [comment]
Let's not commingle fixes. Rename the "installpsh-redhat.sh" script, then patch what it publishes as a .repo file. Those are orthogonal issues and, if the repo file name will be addressed, it really belongs in a different ticket.
